pvlib/solarposition.py

```diff
@@ -1386,22 +1381,26 @@ def hour_angle(times, longitude, equation_of_time):
     equation_of_time_spencer71
     equation_of_time_pvcdrom
     """
+
+    # times must be localized
+    if not times.tz:
+        raise ValueError('times must be localized')
+
     # hours - timezone = (times - normalized_times) - (naive_times - times)
-    if times.tz is None:
-        times = times.tz_localize('utc')
```

Comment:
It was added in accordance with `sun_rise_set_transit_spa` and `sun_rise_set_transit_spa` as well as the docstring. I agree with the previous convention since it prevents users from unwittingly calling the functions assuming an arbitrary timezone or even more naively without any timezones, although an hour angle is counted up from zero at noon in a specific local timezone. Please refer to the followings:

https://github.com/pvlib/pvlib-python/blob/524fa553f3b956a2cf89aeea32548e2a65252f67/pvlib/solarposition.py#L448-L452

https://github.com/pvlib/pvlib-python/blob/524fa553f3b956a2cf89aeea32548e2a65252f67/pvlib/solarposition.py#L560-L564